docs/sphinx/dev_guide/kernel_class_impl.rst

```diff
@@ -52,12 +52,24 @@ The methods in the source file are:
         so these methods are used to describe this.
       * The number of bytes read and written and the number of floating point
         operations (FLOPS) performed for each kernel execution.
-      * The consistency of the checksums of the kernel. The possible values are
-        ``Consistent`` where all the variant tunings always get the same checksum,
-        ``ConsistentPerVariantTuning`` where an individual variant tuning always
-        gets the same checksum but different variant tunings may differ
-        slightly, and ``Inconsistent`` where the checksum of a variant tuning
-        may vary slightly run to run.
+      * The consistency of the checksums of the kernel. If the kernel
+        always produces the same checksum value for all variant tunings then the
+        checksums are ``Consistent``. Most kernels get a different but consistent
+        checksum for each variant tuning so the checksums are
+        ``ConsistentPerVariantTuning``. On the other hand, some kernels have
+        variant tunings that get different checksums on each run of that variant
+        tuning, for example due to the ordering of floating-point atomic add
+        operations, so the checksums are ``Inconsistent``.
```

Comment:
Are ``Inconsistent`` kernels expected to agree within some tolerance? I suspect the ``Inconsistency`` may grow per number of reps?

Reply:
It depends on the kernel, most kernels overwrite their outputs on each rep instead of continuing to accumulate.